content/plugins/pof-importer/pof-importer.php

```diff
@@ -135,309 +187,411 @@ public function import() {
+
+            // Collect connected posts
+            $translations = [];
+            foreach ( $item['languages'] as $lang ) {
+                if ( $lang['page'] ) {
+                    $translations[ $lang['lang'] ] = $lang['page']->ID;
+                    break;
+                }
+            }
+            $new_translations = false;
+
+            // Update translations
+            foreach ( $item['languages'] as &$lang ) {
+                if ( $lang['update_page'] && $lang['data'] ) {
+
+                    // Post object params
+                    $args = [
+                        'post_title'    => sanitize_title( $lang['title'] ),
+                        'menu_order'    => $item['order'],
+                        'post_status'   => 'publish',
+                        'post_type'     => 'page',
+                        'post_title'    => $item['title'],
+                        'post_content'  => $lang['data']['content'],
+                        'page_template' => 'models/page-' . $lang['data']['type'] . '.php',
+                        'post_modified' => $lang['lastModified'],
+                    ];
 
-                    // Default values for page creation/updation
-                    $args['post_status']   = 'publish';
-                    $args['post_type']     = 'page';
-                    $args['post_title']    = $item['title'];
-                    $args['post_parent']   = $parent_id;
-                    $args['post_content']  = isset($item['content']) ? $item['content'] : '';
-                    $args['page_template'] = 'models/page-' . $item['type'] . '.php';
-
-                    
-                    // Create the post
-                    $post_id = wp_insert_post( $args );
-
-                    if ( ! is_wp_error( $post_id ) && $post_id !== 0 && isset( $old_slug ) ) {
-                        $new_slug = wp_make_link_relative( get_permalink( $post_id ));
-                        if ( $new_slug !== $old_slug ) {
-                            $redirect_cache[ $old_slug ] = $new_slug;
+                    // If item has parent link to it
+                    if ( $item['parent'] ) {
+                        $parent_id = $this->queried[ $item['parent'] ][ $lang['lang'] ]->ID;
+                        if ( ! empty( $parent_id ) ) {
+                            $args['post_parent'] = $parent_id;
                         }
                     }
 
+                    // If page already exists update it
+                    if ( $lang['page'] ) {
+                        // Assign id to update page
+                        $args['ID'] = $lang['page']->ID;
+
+                        // Check that there are actually updated values
+                        foreach ( $args as $key => $value ) {
+                            if ( $lang['page']->{ $key } !== $value ) {
+                                // Update the post
+                                $post_id = wp_insert_post( $args );
```

Comment:
Does this run the insert post function multiple times if there a multiple change values? If so, maybe you should optimize this?

Reply:
There is break, doesn't that break the foreach execution?